core/src/main/java/io/grpc/internal/ManagedChannelImpl.java

```diff
@@ -171,7 +171,13 @@ public void onUpdate(List<ResolvedServerInfo> servers, Attributes config) {
         if (servers.isEmpty()) {
           onError(Status.UNAVAILABLE.withDescription("NameResolver returned an empty list"));
         } else {
-          loadBalancer.handleResolvedAddresses(servers, config);
+          try {
+            loadBalancer.handleResolvedAddresses(servers, config);
+          } catch (RuntimeException e) {
+            // It must be a bug! Push the exception back to LoadBalancer in the hope that it may be
+            // propagated to the application.
+            onError(Status.INTERNAL.withCause(e));
```

Comment:
Though I disagree on principle, for consistency we are also catching Error in other places.  Maybe add it here?